test/parallel_api/numeric/numeric.ops/exclusive_scan.pass.cpp

```diff
@@ -26,19 +26,93 @@
+DEFINE_TEST_1(test_with_vector, BinaryOperation)
+{
+    DEFINE_TEST_CONSTRUCTOR(test_with_vector)
+
+    template <typename Iterator1, typename Size>
+    void initialize_data(Iterator1 host_keys, Size n)
+    {
+        for (Size i = 0; i < n; i++)
+        {
+            host_keys[i] = 1;
+        }
+    }
+
+#ifdef DUMP_CHECK_RESULTS
+    template <typename Iterator, typename Size>
+    void display_param(const char* msg, Iterator it, Size n)
+    {
+        std::cout << msg;
+        for (Size i = 0; i < n; ++i)
+        {
+            if (i > 0)
+                std::cout << ", ";
+            std::cout << it[i];
+        }
+        std::cout << std::endl;
+    }
+#endif // DUMP_CHECK_RESULTS
+
+    // specialization for host execution policies
+    template <typename Policy, typename Iterator1, typename Iterator2, typename Iterator3, typename Size>
+    typename ::std::enable_if<
+#if TEST_DPCPP_BACKEND_PRESENT
+        !oneapi::dpl::__internal::__is_hetero_execution_policy<typename ::std::decay<Policy>::type>::value &&
+#endif
+            is_base_of_iterator_category<::std::random_access_iterator_tag, Iterator3>::value,
+        void>::type
+    operator()(Policy&& exec, Iterator1 keys_first, Iterator1 keys_last, Iterator2 res_first, Iterator2 res_last,
+               Iterator3 exp_first, Iterator3 exp_last, Size n)
+    {
+        typedef typename ::std::iterator_traits<Iterator1>::value_type KeyT;
+
+        KeyT init{2};
+
+        initialize_data(keys_first, n);
+        oneapi::dpl::exclusive_scan(std::forward<Policy>(exec), keys_first, keys_last, res_first, init,
+                                    BinaryOperation());
+        exclusive_scan_serial(keys_first, keys_last, exp_first, init, BinaryOperation());
+
+#ifdef DUMP_CHECK_RESULTS
+        display_param("expected: ", exp_first, n);
+        display_param("actual  : ", res_first, n);
+#endif //DUMP_CHECK_RESULTS
+
+        EXPECT_EQ_N(exp_first, res_first, n, "wrong effect from exclusive_scan");
+    }
+
+    // specialization for non-random_access iterators
+    template <typename Policy, typename Iterator1, typename Iterator2, typename Iterator3, typename Size>
+    typename ::std::enable_if<!is_base_of_iterator_category<::std::random_access_iterator_tag, Iterator3>::value,
+                              void>::type
+    operator()(Policy&& exec, Iterator1 keys_first, Iterator1 keys_last, Iterator2 vals_first, Iterator2 vals_last,
+               Iterator3 val_res_first, Iterator3 val_res_last, Size n)
+    {
+    }
+};
+
 #if TEST_DPCPP_BACKEND_PRESENT
 
 #include "support/sycl_alloc_utils.h"
 
-template <sycl::usm::alloc alloc_type>
+template <sycl::usm::alloc alloc_type, typename KernelName, typename _BinaryOp = oneapi::dpl::__internal::__pstl_plus>
 void
-test_with_usm(sycl::queue& q, const ::std::size_t count)
+test_with_usm(sycl::queue& q, const ::std::size_t count, _BinaryOp binary_op = _BinaryOp())
 {
     // Prepare source data
     std::vector<int> h_idx(count);
```

Comment:
Since you changed this to just a vector of 1s, you can skip the loop below and initialize the vector with 1 here.
```c++
std::vector<int> h_idx(count, 1);
```

Reply:
done